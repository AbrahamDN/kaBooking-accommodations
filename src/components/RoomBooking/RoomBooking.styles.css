.room-booking {
  padding: 2rem;
  width: 100%;
}

.room-booking .caption {
  font-weight: 600;
}

@media (min-width: 961px) {
  .room-booking {
    padding: 4rem;
  }
}
