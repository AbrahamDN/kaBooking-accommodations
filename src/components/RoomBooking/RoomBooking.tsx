import "./RoomBooking.styles.css";
import { default as cn } from "classnames";
import { RoomBookingProps } from "./RoomBooking.types.ts";
// import { useRoomStore } from "../../store/room.store";
// import { formatCurrencyString } from "../../lib/formatCurrencyString";

const RoomBooking = ({ className }: RoomBookingProps) => {
  // const { bed } = useRoomStore();
  // const room = mockBed || bed;

  return (
    <article className={cn("room-booking", className)}>
      <div className="">
        {/* <span className="h6">
          {room?.price
            ? formatCurrencyString({
                val: 100,
                currency: room.price.currency_iso_code,
              })
            : "Unavailable"}
        </span> */}
        <span className="caption"> /night</span>
      </div>
    </article>
  );
};

export default RoomBooking;
