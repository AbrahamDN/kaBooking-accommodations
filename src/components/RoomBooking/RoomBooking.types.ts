import { Room } from "../../types/accommodation.types.ts";

export type RoomBookingProps = {
  className?: string;
  mockBed?: Room; // For testing
};
