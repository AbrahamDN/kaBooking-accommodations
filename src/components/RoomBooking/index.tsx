import RoomBooking from "./RoomBooking";

export default RoomBooking;
