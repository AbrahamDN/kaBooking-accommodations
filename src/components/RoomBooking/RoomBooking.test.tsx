import { expect, test } from "vitest";
// import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import RoomBooking from "./RoomBooking";
import toJson from "../../utils/toJson";

test("Room Booking should show", () => {
  const component = renderer.create(<RoomBooking />);

  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
