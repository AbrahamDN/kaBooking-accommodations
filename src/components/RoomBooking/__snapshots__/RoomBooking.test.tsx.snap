// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Room Booking should show 1`] = `
<div
  className="room-booking"
>
  <h3>
    Room Booking
  </h3>
  <label>
    Number of Nights:
    <input
      onChange={[Function]}
      type="number"
      value={5}
    />
  </label>
</div>
`;

exports[`should show title all the time 1`] = `
<div
  className="room-booking"
>
  <h3>
    Room Booking
  </h3>
  <label>
    Number of Nights:
    <input
      onChange={[Function]}
      type="number"
      value={5}
    />
  </label>
</div>
`;
